jsonapi/jsonapi_utility.go

```diff
@@ -167,6 +169,10 @@ func JSONErrorResponse(obj interface{}, err error) error {
 			return errs.WithStack(ctx.Conflict(jsonErr))
 		}
 	default:
+		// report this unknown error to sentry
+		c, _ := raven.New(os.Getenv("SENTRY_DSN"))
```

Comment:
This should be in main.go or similar, not here. No need to reinit the client for each error. Pluss it needs additional 'global' options setup like 'Environment', 'Loggername', 'Release' etc

Reply:
The problem is that client.Capture*() functions are not thread safe. They share the same context. One context per client. So, it looks like we have the following options:
- Use synchronized error reporting (really not an option IMO)
- Create a new client per error report
- Implement our own sentry lib or contribute some changes to raven-go :)

Reply:
for  'Environment', 'Loggername', 'Release'  is this info globally available and static then we can still do it by `reinit`ing client and setting that info by reading from global state/vars

WDYT?

Reply:
If we are not going to change the context between calls then yes, it should work. But I don't see much value if we just report the error stack without any request context.

Reply:
I think you misunderstood me, I was asking about initializing client in main vs doing it on every request? If the info that @aslakknutsen wants to see is static and available globally we can reinit on each request without issues.

@aslakknutsen if we are initializing once and again putting mutexes around it later to serialize access to it, how bad is it to reinitialize the client on each request?

Reply:
We should do number 3, but I'm fine with number 1 for now. 

The mutex is only spanning 'extract info from token, create a capture event, clear the context', the actual 'send to server' is still async. And this is only when a 500 happens. It has occurred in prod 194 times over the past 7 days.